packages/eslint-plugin/lib/rules/require-closing-tags.js

```diff
@@ -37,6 +37,12 @@ module.exports = {
           allowSelfClosingCustom: {
             type: "boolean",
           },
+          customPatterns: {
+            type: "array",
+            items: {
+              type: "string",
+            },
+          },
```

Comment:
i don't think "all custom components" has a valid definition here: without the user defining what the custom components are, the rule can only make assumptions about what they look like

it would be possible to collapse `allowSelfClosingCustom` and `customPatterns` into a single option that just accepts the regex array of custom patterns though, and then treat the options like:
- `selfClosing`: whether void/custom elements should be self closing
- `customPatterns`: which elements are included in `selfClosing` checks

this is probably more intuitive, and would address the ambiguous interpretation of the rule pointed out in your other comment here https://github.com/yeonjuan/html-eslint/pull/188/files/fc25c1c51e9cd1a74950661d8f0f70d829ecbcfb#r1626087825

would collapsing the options be preferred, or is there a good reason to keep the configuration for void tags and custom tags separate?

Reply:
>  is there a good reason to keep the configuration for void tags and custom tags separate?

Yes, because the two options are semantically different. void elements are elements that can't have children, while custom tags can have children depending on the implementation.

Reply:
> void elements are elements that can't have children, while custom tags can have children depending on the implementation

this is captured by the current implementation: if self-closing is enforced, but a custom tag has children, it won't be considered an error ([test case](https://github.com/yeonjuan/html-eslint/pull/188/files#diff-e05ac20cc51381258425f49ddb836fe746674ec83a8fbbf61683d42d6b47eba6R65))

i think i'm not quite following the intent here, so to clarify: my assumption was that custom elements should match the prescribed behaviour for void elements (i.e. they should either enforce self-closing, or prevent self-closing), and an exception was made for when they have children (since it's unknown whether children are valid, and this may be independent of whether the tag can self-close).

is the actual intent as follows?

- prescribe whether void elements should self-close or not
- prescribe a set of custom elements which should self-close
- prescribe a set of custom elements which should never self-close

i can update the implementation to reflect that if so, but i don't think the current option names are clear if that's the case

Reply:
> prescribe a set of custom elements which should self-close
prescribe a set of custom elements which should never self-close

I think by default custom tags should not be able to use self closing, but only allow tags specified via options.

why?
It is a non-standard HTML code to use self-closing on tags that are not void-elements. Nevertheless, the selfClosing option was added for convenience when used with the template engine. So I think using selfClosing in a custom tag is a “exceptional” cases. 
That's why I don't think there's much point in forcing selfClosing for a given custom element.

- https://github.com/yeonjuan/html-eslint/issues/99#issue-1260905076

Reply:
> prescribe whether void elements should self-close or not
prescribe a set of custom elements which should self-close
prescribe a set of custom elements which should never self-close

👍 

@seleb What about the selfClosingCustomPatterns option? I was thinking of forcing self-closing if the regular expression in selfClosingCustomPatterns is matched (if there are no children), and disabling self-closing otherwise.

```
selfClosing: never | always
selfClosingCustom: ["-"]
```

Reply:
using `selfClosingCustomPatterns` that way makes sense to me, i'll try to update this later today to meet that spec